dev/install-common-deps.sh

```diff
@@ -48,8 +55,10 @@ if [[ "$INSTALL_LARGE_PYTHON_DEPS" == "true" ]]; then
   ls -lha $(find $CONDA_DIR/envs/test-environment/ -path "*bin/spark-*")
 fi
 
-pip install .[extras]
-export MLFLOW_HOME=$(pwd)
+# Skinny Client purposefully removes the sqlalchemy dependency
+if [[ "$MLFLOW_SKINNY" == "true" ]]; then
+  pip uninstall sqlalchemy -y
+fi
```

Comment:
@eedeleon How do we know that installing MLflow Skinny didn't pull in `sqlalchemy` independently of the Python small dependencies? We could change the ordering here so that the MLflow client is installed after the small dependencies & removal of `sqlalchemy`, but I'd rather create a `skinny-requirements` file rather than rely on subtle ordering properties / environment-variable-based branching logic in `install-common-deps`.

Reply:
I made the change. There is some overlap between small-requirements and large requirements, but I agree it is worth it since this approach is definitely more explicit and less likely to leak bugs.